sources/osgShadow/ShadowMap.js

```diff
@@ -420,45 +420,45 @@ define( [
 
             this._shadowReceiveAttribute.setLightNumber( lightNumber );
 
-            this._receivingStateset.setAttributeAndModes( this._shadowReceiveAttribute, StateAttribute.ON | StateAttribute.OVERRIDE );
 
 
+            var fullOverride = StateAttribute.OVERRIDE | StateAttribute.ON;
 
-            // prevent unnecessary texture bindings
 
-            if ( !ShadowMap.BlankTexture ) {
-                ShadowMap.BlankTexture = new Texture();
-                ShadowMap.BlankTexture.setName( 'emptyTex' );
-            }
-            var blankTexture = ShadowMap.BlankTexture;
+            this._receivingStateset.setAttributeAndModes( this._shadowReceiveAttribute, fullOverride );
+
 
-            // Mandatory: prevent binding shadow textures themselves (shadowedScene stateSet is applied
-            // just above in stateset hierarchy
-            // that would mean undefined values as it would be read/write access...
-            this._casterStateSet.setTextureAttributeAndModes( this._textureUnit, blankTexture, StateAttribute.OFF | StateAttribute.OVERRIDE | StateAttribute.PROTECTED );
+
+            // prevent unnecessary texture bindings
+            var blankTexture = Texture.textureNull;
 
             // TODO: optimize: have a "don't touch current Texture stateAttribute"
             // on ALL texture unit but alpha max texture for Depth/Normal/etc renders
 
             // TODO: actually get the real max texture unit from webglCaps
-            //var shouldGetMaxTextureUnits = 32;
+            var shouldGetMaxTextureUnits = 32;
             //
-            // TODO:: Should handle the alpha_mask that uses a texture
-            // case somehow...
+            // TODO:: Should handle the alpha_mask that uses a texture case somehow...
             // deduce from shader compil ?
-            //for ( var k = 0; k < shouldGetMaxTextureUnits; k++ ) {
-            //    this._casterStateSet.setTextureAttributeAndModes( k, blankTexture, StateAttribute.OFF | StateAttribute.OVERRIDE | StateAttribute.PROTECTED );
-            //}
+            for ( var k = 0; k < shouldGetMaxTextureUnits; k++ ) {
```

Comment:
maybe split this part into a function that says what it does, just to understand better. This part is to 'optimize' avoid texture binding right ? so maybe this could be done in a function with an appropriate name